refinery/tool_manager/utils.py

```diff
@@ -375,3 +427,56 @@ def get_visualization_annotations_list():
             visualization_annotations.append(json.loads(f.read()))
 
     return visualization_annotations
+
+
+def parse_file_relationship_nesting(nested_structure, nesting_dict=None,
+                                    nesting_level=0):
+    """
+    Recursive method to ensure that a ToolLaunchConfiguration's
+    `file_relationships` string is a proper representation of a LIST/PAIR
+    structure
+    :raises: RuntimeError if an inappropriately configured `file_relationships`
+     string is detected
+    """
+
+    if nesting_dict is None:
```

Comment:
I think its okay to keep them separate here. 

The  first `if` is just instantiating the `nesting_dict` to be reused in recursive calls.

Since each `nesting_level` could have many objects, I only want to set the `nesting_level` key once, which is why the KeyError check is helpful.